pinot-common/src/main/java/org/apache/pinot/common/function/FunctionInfo.java

```diff
@@ -19,17 +19,27 @@
 package org.apache.pinot.common.function;
 
 import java.lang.reflect.Method;
+import org.apache.calcite.schema.Function;
+import org.apache.calcite.schema.impl.ScalarFunctionImpl;
+import org.apache.pinot.spi.annotations.ScalarFunction;
 
 
 public class FunctionInfo {
   private final Method _method;
   private final Class<?> _clazz;
   private final boolean _nullableParameters;
 
-  public FunctionInfo(Method method, Class<?> clazz, boolean nullableParameters) {
-    _method = method;
-    _clazz = clazz;
-    _nullableParameters = nullableParameters;
+  public FunctionInfo(Function calciteFunction) {
```

Comment:
We don't want to do reflection on a per function per query basis. One way to avoid it is to have a wrapper class over `Function` and `FunctionInfo`, where `FunctionInfo` is pre-constructed